pkg/cmd/dockerregistry/dockerregistry.go

```diff
@@ -242,3 +248,31 @@ func panicHandler(handler http.Handler) http.Handler {
 		handler.ServeHTTP(w, r)
 	})
 }
+
+func setDefaultMiddleware(config *configuration.Configuration) (changed bool, err error) {
+	// Default to openshift middleware for relevant types
+	// This allows custom configs based on old default configs to continue to work
+	if config.Middleware == nil {
+		config.Middleware = map[string][]configuration.Middleware{}
+	}
+	for _, middlewareType := range []string{"registry", "repository", "storage"} {
+		if _, exists := config.Middleware[middlewareType]; !exists {
+			config.Middleware[middlewareType] = []configuration.Middleware{}
+		}
+		found := false
+		for _, middleware := range config.Middleware[middlewareType] {
+			if middleware.Name == "openshift" {
+				found = true
+				break
+			}
+		}
+		if found {
+			continue
+		}
+		config.Middleware[middlewareType] = append(config.Middleware[middlewareType], configuration.Middleware{
```

Comment:
Is there any case where replacing the openshift registry, repository, or storage middleware would be what someone wants, or is our registry unable to run unless all three openshift middlewares are configured? If someone does have custom ones registered, does appending openshift after theirs always make sense? (I don't know what impact ordering has here)